extractor/image/daemon/image.go

```diff
@@ -0,0 +1,134 @@
+package daemon
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"sync"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"golang.org/x/xerrors"
+)
+
+var mu sync.Mutex
+
+// image is a wrapper for github.com/google/go-containerregistry/pkg/v1/daemon.Image
+// daemon.Image loads the entire image into the memory at first,
+// but it doesn't need to load it if the information is already in the cache,
+// To avoid entire loading, this wrapper uses ImageInspectWithRaw and checks image ID and layer IDs.
+type image struct {
+	v1.Image
+	opener  opener
+	inspect types.ImageInspect
+}
+
+type opener func() (v1.Image, error)
+
+func Image(ref name.Reference) (v1.Image, func(), error) {
```

Comment:
We should add some documentation here saying that `cleanup` func must be called by the caller of this function. Otherwise we'll be filling up precious disk space.

Reply:
I added. https://github.com/aquasecurity/fanal/pull/96/commits/8bdb3a5920606dfb68ae433f859092a3b1bb21a0